### ISSUE TITLE
Error when resolving trait associated types with higher order lifetime

### ISSUE THREAD

**Issue by jialunzhang-psu** [rustanlys/rupta#2]
For such a mir assignment,

```rust
_4 = Option::<for<'a> fn(&'a ()) -> <S2 as P<'a>>::D>::None;
```

Rupta throws an error saying "`TraitPredicate(<S2 as P<'_>>, polarity:Positive)` has escaping bound vars, so it cannot be wrapped in a dummy binder." Looks like the `for<'a>` lifetime bound was not taken into account, causing a free lifetime `'a`.

For a minimal reproducible example, see [this](https://gist.github.com/jialunzhang-psu/978422cb6b459f44623e10cac09cabf6).

**Comment by rustanlys** [comment]
Fixed in commit [aff2893](https://github.com/rustanlys/rupta/commit/aff28930957bcc4ae59d7c4e9e7746f1b767d938).

**Comment by rustanlys** [comment]
Fixed in commit [aff2893](https://github.com/rustanlys/rupta/commit/aff28930957bcc4ae59d7c4e9e7746f1b767d938).
